R/period.R

```diff
@@ -402,8 +402,18 @@ yearquarter.Date <- yearquarter.POSIXt
 yearquarter.character <- function(x) {
   if (is_empty(x)) return(as_yearquarter(x))
 
-  anytime::assertDate(x)
-  as_yearquarter(anytime::anydate(x))
+  qtr_regex <- "(?i)(q|qtr|quarter)(?-i) *"
+  if(all(grepl(paste0("\\d{4} *", qtr_regex, "\\d|", qtr_regex, "\\d *\\d{4}"), x, perl = TRUE))){
```

Comment:
1. Can you put space around the `if-else` statement?
2. In the condition, can you just do `grepl(qtr_regex, x, perl = TRUE)` without checking other things?

Reply:
1. Fixed
2. Why? It's safer to match the whole expression, rather than accept an incomplete expression.